### ISSUE TITLE
Better feedback for failed sync rules

### ISSUE THREAD

**Issue by llermaly** [elastic/connectors-python#1523]
### Problem Description
When many rules fail, it is hard to identify which rule failed or why.

### Proposed Solution
1. Add more details about which rule failed
2. Refer to the rule number, or highlight the rule that failed. Currently the error refers to an ID which is not present in the UI

### Additional Context
![CleanShot 2023-08-23 at 18 18 54](https://github.com/elastic/connectors-python/assets/8575569/da8f70a8-5fc2-44e4-8ae2-271d68689391)

For example on this screenshot all the errors are triggering for rule #12 ?

**Comment by danajuratoni** [comment]
We should not use IDs in UI error messages that are not exposed in Kibana, as [previously](https://github.com/elastic/enterprise-search-team/issues/3482#issuecomment-1336990506) discussed.

Referencing the rule with relative numbering e.g. rule 2 and 4 conflict, should have been the go to approach since several versions now, marking this as a bug that should be fixed asap.

Ideally, we should be highlighting the affected rows (using IDs internally) and display only a user friendly/meaningful error message in Kibana. @efegurkan could you please check if the highlighting logic was implemented (I believe 8.8 was when we originally planned this for)

**Comment by danajuratoni** [comment]
cc: @artem-shelkovnikov

**Comment by seanstory** [comment]
This isn't going to make 8.10.0

**Comment by timgrein** [comment]
I've adjusted the effort to `medium`. This requires a small protocol change adding a `order` field. We could solve it in a hacky way setting the `id` to the `order`, but that would make it hard to debug it later as the `order` can be changed, but the `id` not.

**Comment by DianaJourdan** [comment]
@timgrein what is the remaining work on this bug? Just the Kibana header change? "additional Kibana change is needed to also adapt the heading of the error message in the UI."

**Comment by seanstory** [comment]
> This requires a small protocol change adding a order field.

I don't think that's the case. Lists/arrays are ordered in JSON - you can use the index in the list/array to know what order the rule is in. 

But regardless, the way I'd looked at addressing this (and still think it could be handled) is to just improve the `to_string` function for these rules, so that it's more clear to the user what rules are conflicting.

**Comment by DianaJourdan** [comment]
Agreed in the team sync that no protocol change is needed, but we will stick to an error message + highlight the rules that fail
